pkg/selector/outputs/bubbletea.go

```diff
@@ -83,52 +117,90 @@ func (m BubbleTeaModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 	switch msg := msg.(type) {
 	case tea.KeyMsg:
 		switch msg.String() {
-		case "ctrl+c", "esc", "q":
+		case "ctrl+c", "q":
 			return m, tea.Quit
+		case "enter":
+			switch m.state {
+			case stateTable:
+				// switch from table state to verbose state
+				m.state = stateVerbose
+
+				// get focused instance type
+				rowIndex := m.TableModel.GetHighlightedRowIndex()
+				focusedInstance := m.verboseView.instanceTypes[rowIndex]
+
+				// set content of view
+				m.verboseView.focusedInstanceName = focusedInstance.InstanceType
+				m.verboseView.viewport.SetContent(VerboseInstanceTypeOutput([]*instancetypes.Details{focusedInstance})[0])
+
+				// move viewport to top of printout
+				m.verboseView.viewport.SetYOffset(0)
+			case stateVerbose:
+				// switch from verbose state to table state
+				m.state = stateTable
+			}
 		}
 	case tea.WindowSizeMsg:
 		// handle screen resizing
-
-		// This is needed to handle a bug with bubble tea
-		// where resizing causes misprints (https://github.com/Evertras/bubble-table/issues/121)
-		termenv.ClearScreen()
-
-		// handle width changes
-		m.TableModel = m.TableModel.WithMaxTotalWidth(msg.Width)
-
-		// handle height changes
-		if headerAndFooterPadding >= msg.Height {
-			// height too short to fit rows
-			m.TableModel = m.TableModel.WithPageSize(0)
-		} else {
-			newRowsPerPage := msg.Height - headerAndFooterPadding
-			m.TableModel = m.TableModel.WithPageSize(newRowsPerPage)
-		}
+		m = resizeTableView(m, msg)
+		m = resizeVerboseView(m, msg)
 	}
 
-	// update table
-	var cmd tea.Cmd
-	m.TableModel, cmd = m.TableModel.Update(msg)
-
-	// update footer
-	controlsStr := lipgloss.NewStyle().Faint(true).Render(controlsString)
-	footerStr := fmt.Sprintf("Page: %d/%d | %s", m.TableModel.CurrentPage(), m.TableModel.MaxPages(), controlsStr)
-	m.TableModel = m.TableModel.WithStaticFooter(footerStr)
+	switch m.state {
+	case stateTable:
+		// update table
+		var cmd tea.Cmd
+		m.TableModel, cmd = m.TableModel.Update(msg)
+
+		// update footer
+		controlsStr := lipgloss.NewStyle().Faint(true).Render(tableControls)
+		footerStr := fmt.Sprintf("Page: %d/%d | %s", m.TableModel.CurrentPage(), m.TableModel.MaxPages(), controlsStr)
+		m.TableModel = m.TableModel.WithStaticFooter(footerStr)
+
+		return m, cmd
+	case stateVerbose:
+		// update viewport
+		var cmd tea.Cmd
+		m.verboseView.viewport, cmd = m.verboseView.viewport.Update(msg)
+		return m, cmd
+	}
 
-	return m, cmd
+	return m, nil
 }
 
 // View is used by bubble tea to render the bubble tea model
 func (m BubbleTeaModel) View() string {
 	outputStr := strings.Builder{}
 
-	outputStr.WriteString(m.TableModel.View())
-	outputStr.WriteString("\n")
+	switch m.state {
+	case stateTable:
+		outputStr.WriteString(m.TableModel.View())
+		outputStr.WriteString("\n")
+	case stateVerbose:
+		// format header for viewport
```

Comment:
Our component-specific logic is getting long enough that I'd suggest we move the components into their own files i.e. `verboseView.go`, `table.go`, and you can the logic here and other places to look something like: 
```
...
switch m.state {
    case stateTable:
        return m.table.View()
    case stateVerbose:
        return m.verboseView.View()    
}
...
```